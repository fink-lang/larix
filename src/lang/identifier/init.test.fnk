{describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'

{parse_expr} = import '../../parser.fnk'


describe 'indentifiers', fn:
  it 'parses a word', fn:
    expect
      parse_expr '  foobar  '
      to_match_snapshot


  it 'parses a non-word', fn:
    expect
      parse_expr ' Ï€ '
      to_match_snapshot

  it 'parses a hyphenated', fn:
    expect
      parse_expr ' foo-bar '
      to_match_snapshot
