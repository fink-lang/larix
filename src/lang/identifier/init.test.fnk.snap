// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`indentifiers parses a hyphenated 1`] = `ident (1:1-1:8) foo-bar`;

exports[`indentifiers parses a non-word 1`] = `ident (1:1-1:2) Ï€`;

exports[`indentifiers parses a word 1`] = `ident (1:2-1:8) foobar`;
